annotations/src/main/java/com/squareup/anvil/annotations/ContributesBinding.kt

```diff
@@ -98,11 +103,36 @@ public annotation class ContributesBinding(
    * use the same scope.
    */
   val replaces: Array<KClass<*>> = [],
+  /**
+   * The priority of this contributed binding. The priority should be changed only if you don't
+   * have access to the contributed binding class that you want to replace at compile time. If
+   * you have access and can reference the other class, then it's highly suggested to
+   * use [replaces] instead.
+   *
+   * In case of a duplicate binding for multiple contributed bindings the binding with the highest
+   * priority will be used and replace other contributed bindings for the same type with a lower
+   * priority. If duplicate contributed bindings use the same priority, then there will be an
+   * error for duplicate bindings.
+   *
+   * Note that [replaces] takes precedence. If you explicitly replace a binding it won't be
+   * considered no matter what its priority is.
+   *
+   * All contributed bindings have a [NORMAL] priority by default.
+   */
+  val priority: Priority = NORMAL,
   /**
    * Whether the qualifier for this class should be included in the generated binding method. This
    * parameter is only necessary to use when [ContributesBinding] and [ContributesMultibinding]
    * are used together for the same class. If not, simply remove the qualifier from the class
    * and don't use this parameter.
    */
   val ignoreQualifier: Boolean = false
-)
+) {
+  /**
+   * The priority of a contributed binding.
+   */
+  @Suppress("unused")
+  public enum class Priority {
+    NORMAL, HIGH, HIGHEST
```

Comment:
LOW, NORMAL, HIGH would make more sense to me. Normal would be the default. Low would be for fake/mock objects. High would be for those rare situations where you need to override the normal priority.

Why'd you decide to go with enums in the end instead of ints?

Reply:
In the end I decided for an enum for better auto-complete. And I really can't think of a scenario where you need more options and can't rely on `replaces` either.

Hm, I preferred these three constants, because my gut thinks that you rather want to override existing bindings than provide one with lower priority. I don't want to use `LOW` as the default, because that is awkward to read.

Reply:
I see what you mean. I'm fine with the way it is then.